### ISSUE TITLE
Tests failing to compile

### ISSUE THREAD

**Issue by mkfifo** [carp-lang/Carp#212]
```
$ stack test

CarpHask-0.2.0.0: build (lib + exe + test)
Preprocessing library CarpHask-0.2.0.0...
Preprocessing executable 'carp' for CarpHask-0.2.0.0...
Preprocessing executable 'carp-header-parse' for CarpHask-0.2.0.0...
Preprocessing test suite 'CarpHask-test' for CarpHask-0.2.0.0...
[1 of 1] Compiling Main             ( test/Spec.hs, .stack-work/dist/x86_64-linux-tinfo6-nopie/Cabal-1.24.2.0/build/CarpHask-test/CarpHask-test-tmp/Main.o )
             
/home/chris/devel/Carp/test/Spec.hs:33:10: error:
    • Couldn't match expected type ‘Obj’
                  with actual type ‘Maybe String -> Obj’
    • Probable cause: ‘External’ is applied to too few arguments
      In the first argument of ‘XObj’, namely ‘External’
      In the expression: XObj External Nothing Nothing
      In an equation for ‘x’: x = XObj External Nothing Nothing
Progress: 1/2
--  While building custom Setup.hs for package CarpHask-0.2.0.0 using:
      /home/chris/.stack/setup-exe-cache/x86_64-linux-tinfo6-nopie/Cabal-simple_mPHDZzAJ_1.24.2.0_ghc-8.0.2 --builddir=.stack-work/dist/x86_64-linux-tinfo6-nopie/Cabal-1.24.2.0 build lib:CarpHask exe:carp exe:carp-header-parse test:CarpHask-test --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1

```

I'm also able to reproduce this on travis-ci
https://travis-ci.org/mkfifo/Carp/builds/360484300

**Comment by mkfifo** [comment]
PR to fix at #213

**Comment by mkfifo** [comment]
Fixed.
